### PR TITLE
feat(rpc): report JWT path in case of an IO read error

### PR DESCRIPTION
**Pull request by shekhirin**
In case when user mistakenly specifies a directory as JWT secret path:
```console
➜  reth (alexey/jwt-io-error-path) mkdir jwt
➜  reth (alexey/jwt-io-error-path) cargo run -- node --authrpc.jwtsecret $(pwd)/jwt
```

We don't mention the path in the error message:
```console
Error: An I/O error occurred: Is a directory (os error 21)

Caused by:
    Is a directory (os error 21)

Location:
    bin/reth/src/node/mod.rs:376:26
```

This PR fixes it:
```console
Error: IO error occurred while reading /Users/shekhirin/Projects/reth/jwt: Is a directory (os error 21)

Location:
    bin/reth/src/node/mod.rs:376:26
```

### CODE REVIEW

File: crates/rpc/rpc/src/layers/jwt_secret.rs [hunk 1]
```diff
@@ -26,6 +26,8 @@ pub enum JwtError {
     MissingOrInvalidAuthorizationHeader,
     #[error("JWT decoding error {0}")]
     JwtDecodingError(String),
+    #[error("IO error occurred while reading {path}: {err}")]
```
Comment (mattsse) [inline crates/rpc/rpc/src/layers/jwt_secret.rs:29]: "reading" is probably a sufficient context that we interpreted this as a file, I think

File: crates/rpc/rpc/src/layers/jwt_secret.rs [hunk 2]
```diff
@@ -92,7 +95,8 @@ impl JwtSecret {
         let secret = JwtSecret::random();
         let bytes = &secret.0;
         let hex = hex::encode(bytes);
-        std::fs::write(fpath, hex)?;
+        std::fs::write(fpath, hex)
+            .map_err(|err| JwtError::IORead { err, path: fpath.to_path_buf() })?;
```
Comment (mattsse) [inline crates/rpc/rpc/src/layers/jwt_secret.rs:99]: this error variant does not match the operation

**Review by mattsse — COMMENTED** [review]
some early comments,

this is always terrible, it's always the same code -.-


FYI: https://github.com/foundry-rs/foundry/blob/master/common/src/fs.rs

**Comment by shekhirin** [comment]
> FYI: https://github.com/foundry-rs/foundry/blob/master/common/src/fs.rs

sounds like a good good first issue

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**